cmd/kubelet-plugin/pci_device.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/google/uuid"
+)
+
+const (
+	pciBasePath = "/sys/bus/pci/devices"
+	PCIResourcePrefix   = "PCI_RESOURCE"
+)
+
+type PCIDevice struct {
+	uuid           string
+	vendorSelector string
+	resourceName   string
+	pciAddress     string
+	driver         string
+	iommuGroup     string
+	numaNode       int
+	pciID          string
+}
+
+func DiscoverPermittedHostPCIDevices(supportedPCIDeviceMap map[string]string) (map[string][]*PCIDevice) {
+	initHandler()
+
+	iommuToPCIMap := make(map[string]string)
+
+	pciDevicesMap := make(map[string][]*PCIDevice)
+	err := filepath.Walk(pciBasePath, func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() {
+			return nil
+		}
+		pciID, err := Handler.GetDevicePCIID(pciBasePath, info.Name())
+		if err != nil {
+            log.Printf("Failed to get vendor:device ID for device: %s, error: %v", info.Name(), err)
+			return nil 
+		}
+
+		if _, supported := supportedPCIDeviceMap[pciID]; supported {
+			driver, err := Handler.GetDeviceDriver(pciBasePath, info.Name())
+			if err != nil || driver != "vfio-pci" {
+				log.Printf("Driver error: %v", err)
+				return nil 
+			}
+
+			pcidev := &PCIDevice{
+				uuid:           uuid.New().String(),
+				pciID:          pciID,
+				pciAddress:     info.Name(),
+				vendorSelector: pciID,
+				resourceName:   supportedPCIDeviceMap[pciID],
+			}
+
+			iommuGroup, err := Handler.GetDeviceIOMMUGroup(pciBasePath, info.Name())
+			if err != nil {
+				log.Printf("IOMMU group error: %v", err)
+                return nil
+			}
+			pcidev.iommuGroup = iommuGroup
+			pcidev.driver = driver
+			pcidev.numaNode = Handler.GetDeviceNumaNode(pciBasePath, info.Name())
+
+			iommuToPCIMap[pcidev.iommuGroup] = pcidev.pciAddress
+
+			pciDevicesMap[pciID] = append(pciDevicesMap[pciID], pcidev)
+		}
+		return nil
+	})
+	if err != nil {
+        log.Printf("Failed to discover host devices, error: %v", err)
+	}
+
+	return pciDevicesMap
+}
+
+// MockDiscoverPermittedHostPCIDevices returns predefined data for testing
+func MockDiscoverPermittedHostPCIDevices(supportedPCIDeviceMap map[string]string) (map[string][]*PCIDevice, map[string]string) {
```

Comment:
The signature of this function is different from the function `DiscoverPermittedHostPCIDevices`, is it done on purpose?
```golang
DiscoverPermittedHostPCIDevices(supportedPCIDeviceMap map[string]string) (map[string][]*PCIDevice)
```